contracts/MechMarketplace.sol

```diff
@@ -744,80 +824,15 @@ contract MechMarketplace is IErrorsMarketplace {
         _locked = 2;
 
         // Array length checks
-        if (requestDatas.length == 0 || requestDatas.length != signatures.length ||
-            requestDatas.length != deliveryDatas.length || requestDatas.length != deliveryRates.length) {
-            revert WrongArrayLength4(requestDatas.length, signatures.length, deliveryDatas.length, deliveryRates.length);
+        if (deliverWithSignatures.length == 0 || deliverWithSignatures.length != deliveryRates.length) {
+            revert WrongArrayLength(deliverWithSignatures.length, deliveryRates.length);
         }
 
-        // Check mech
-        address mechServiceMultisig = checkMech(msg.sender);
```

Comment:
The rest of the function is CODE UNCHANGED, just above in the internal extracted one.